Base

senago opened the pull request.

senago committed:
Impelement the base

senago committed:
Update README.md

senago committed:
Refactor and add Handlebars

senago committed:
Refactor

senago committed:
Refactor

senago committed:
Love JS so much! Won't ever switch to TSgit add .

senago committed:
Build basic architecture

senago committed:
Run prettier

senago committed:
Implement Signup View

senago committed:
Update .gitignore

senago committed:
Refactor

senago committed:
Configure and run ESLint

senago committed:
Remove ESLint and rename Makefile

senago committed:
Remove cookies, add singletone logic

senago committed:
Refactor Registries and Router

senago committed:
Fix ESLint errors

senago committed:
Fix Router bug

senago committed:
Change 'sh' to 'bash' for Ubuntu compatibility

senago committed:
Change BackendBaseURL

senago committed:
Enable input validation

senago committed:
Manual merge with develop-feed

senago committed:
Add login and get user data methods

senago committed:
Switch to nginx

senago committed:
Fix bugs

senago committed:
Remove unnecessary export

senago committed:
Stage

senago committed:
Add logout logic

senago committed:
Fix auth method

Similization committed:
feed commit

senago committed:
Merge branch 'develop-auth' into stage

senago committed:
Cleanup

senago committed:
Fix bug

senago committed:
Hotfix

senago committed:
Start refactoring

senago committed:
Refactor Component model and remove unnecessary files

senago committed:
Refactor View model

senago committed:
Remove unnecessary node js dep from Dockerfile

senago committed:
Add basic adapters

senago committed:
Fix nginx bug

senago committed:
Add adapters logic to views

senago committed:
Add state check before view render

senago committed:
Update README

senago committed:
Refactor error generation

senago committed:
Refactor Router and fix SignupView bug

senago committed:
Refactor adapters call in View

senago committed:
Merge pull request #2 from frontend-park-mail-ru/stage

Stage

senago committed:
Minor fix

senago committed:
Fix several bugs and enable JIT local mode

senago committed:
Enable quiet mode in inotifywait

senago committed:
Refactor, add css packing, start implementing design system

senago committed:
Implement some components of the design system

senago committed:
Turn Signup Form id back

senago committed:
Minor fix

senago committed:
Improve design system, add login view, add Router actions

senago committed:
Run eslint

senago committed:
Add helpers

senago committed:
Add dirty validation

senago committed:
Minor fix

senago committed:
Fix Dockerfile

senago committed:
Fix bug

senago committed:
Refactor

senago committed:
Choose better naming

senago committed:
Add runner for mac os

senago committed:
Fix

senago committed:
Fix Docker run

senago committed:
Refactor EventBus and Services

senago committed:
Run eslint

senago committed:
Refactor Validation and Component model

senago committed:
Minor fix

senago committed:
Fix bug

senago committed:
Fix some bugs

senago committed:
Choose better naming

senago committed:
Improve readability

senago committed:
Updage .gitignore

senago committed:
Get rid of controllers

senago committed:
Update README

senago committed:
Run eslint

senago committed:
Small fixes

senago committed:
Add data binding to docker

senago committed:
Implement Store

senago committed:
Refactor adapters

senago committed:
Refactor layout and Store

senago committed:
Remove redundant log

senago committed:
Setup and run prettier

senago committed:
Cleanify

senago committed:
Refactor input validation

senago committed:
Major refactor of Router: add Controllers

senago committed:
Turn off password requirements

senago committed:
Add links to header

senago committed:
Add feed mock

senago committed:
Get rid of state check in View

senago committed:
Change layout of Controllers

senago committed:
Change layout of Stores

senago committed:
Refactor Component (add separate Context)

senago committed:
Segregate Action

senago committed:
Get rid of email type inputs

senago committed:
Move middlewares to controllers dir

senago committed:
Move middlewares to stores dir

senago committed:
Refactor controllers

senago committed:
Slight refactor of ComponentsRegistry, start implementing Feed

senago committed:
Get rid of EventBus usage

senago committed:
Remove EventBus module

senago committed:
Refactor registries

senago committed:
Change layout of views registry

senago committed:
Change layout of components registry

senago committed:
Cleanup

senago committed:
Update configs

senago committed:
Major refactor of components and views

senago committed:
Cleanup, decouple

senago committed:
Stage

senago committed:
Add binding to backend network

Similization committed:
stage

Similization committed:
clean feed

Similization committed:
likes, comment, replys (only first post)

Similization committed:
add local sources and normal likes count

senago committed:
Major refactor and cleanup

senago committed:
Choose better naming

senago committed:
Cleanup store

senago committed:
Choose better naming

senago committed:
Merge branch 'base-develop' into base

Similization committed:
added profile components

Similization committed:
added profile components

Similization committed:
rebase

Similization committed:
Profile correction

alexey-ershkov commented on .eslintrc.js:
```diff
@@ -0,0 +1,18 @@
+    es2021: true
+  },
+  extends: ['eslint:recommended'],
+  // extends: ['eslint:recommended', 'plugin:import/recommended'],
```
Кажется это можно убрать

alexey-ershkov commented on Dockerfile:
```diff
@@ -0,0 +1,8 @@
+
+EXPOSE 80
+
+# CMD ["nginx", "-g", "daemon off;"]
```
??

alexey-ershkov commented on src/components/LoginForm/LoginForm.js:
```diff
@@ -0,0 +1,36 @@
+import { createComponent } from '../../core/models/Component/Component.js';
+import { InputIDs, InputsRegistry, InputTypes } from '../../core/modules/InputValidator/InputsRegistry.js';
+import { Router } from '../../core/modules/Router/Router.js';
+import { LoginUserDTO } from '../../core/network/dto/auth.js';
```
Опять DTO) Назовите без сокращений

alexey-ershkov commented on src/components/Menu/Menu.css:
```diff
@@ -0,0 +1,18 @@
+/* Menu starts */
```
Кажется лишнее - по названию файла и так понятно

Similization replied:
Убрал

alexey-ershkov commented on src/components/Menu/Menu.js:
```diff
@@ -0,0 +1,3 @@
+import { createComponent } from '../../core/models/Component/Component.js';
+
+export const menuComponent = (template) => createComponent(template);
```
Кажется можно не писать подобный файлики, а сделать общий генератор

Similization replied:
Да, но это не столь необходимо для 2 строчек кода

alexey-ershkov commented on src/components/Post/Post.css:
```diff
@@ -0,0 +1,229 @@
+    grid-gap: 0.5rem;
+}
+
+.info > p:first-child {
```
Опять же опираетесь на порядок и на html элементы

Similization replied:
Это плохо ?

alexey-ershkov commented on src/components/Post/Post.css:
```diff
@@ -0,0 +1,229 @@
+
+.comments-section {
+    display: none;
+    /* animation: 3s ease-in 1s 2 reverse both paused slidein; */
```
???

Similization replied:
Не очевидно

alexey-ershkov commented on src/components/Post/Post.css:
```diff
@@ -0,0 +1,229 @@
+    align-items: unset;
+}
+
+/* .comments-list {
```
Комменты лучше все выпилить

Similization replied:
Комменты к посту или о чем речь ?

alexey-ershkov commented on src/components/Post/Post.handlebars:
```diff
@@ -0,0 +1,123 @@
+      <img src='../sources/icons/comment.svg' alt='' class='icon' />
+      <p>comment</p>
+    </div>
+<<<<<<< HEAD
```
???

Similization replied:
Тут что-то поехало

alexey-ershkov commented on src/components/Post/Post.js:
```diff
@@ -0,0 +1,128 @@
+ */
+const changeLike = (e) => {
+  function simplify(value, numCount) {
+    switch (true) {
```
Что-то страшное - можно вынести в объект, но накиньте еще комментов, что здесь происходит, прям оч странно

Similization replied:
В общем какая была идея
Я вывел +- адекватную формулу для вывода красивых лайков
В чем суть:
Лайки могут иметь следующий вид:
1
12
123
1k
1.23k
12.1k
723k
1.12m
и тд, но мы не рассчитываем на числа большие 999 миллионов, так что все просто

Общем первый if поверяет что число содержит менее 4 чисел -> оно отображается как обычно
Иначе мы его упрощаем
Что это значит ?
Если в числе 4 цифры, то мы делим его на 10**(1 == 4 - 3), округляем и делим на 10**(2 == 3*2 - 4) и получаем красивое число из трех цифр и одной точки (Пример 4782 лайка превратятся в 4.78)
Если в числе 5 цифр, то мы делим его на 10**(2 == 5 - 3), округляем и делим на 10**(1 == 3*2 - 5) и получаем красивое число из трех цифр и одной точки (Пример 47826 лайка превратятся в 47.8)
Если в числе 6 цифр, то мы делим его на 10**(3 == 6 - 3), округляем и делим на 10**(2 == 3*2 - 6) и получаем красивое число из трех цифр и одной точки (Пример 478265 лайка превратятся в 478)

Eсли цифр больше, то мы упрощаем число в соответствии с верхним алгоритмом: кол-во цифр в числе должно стать 4 если кол-во цифр % 3 == 1 
стать 5 если кол-во цифр % 3 == 2
стать 6 если кол-во цифр % 3 == 0
Math.round() округлит число до нужного кол-ва

Затем по старой длине мы определяем переписку 
'k', если число лежит в пределах [1.000; 1.000.000)
'm', если число лежит в пределах [1.000.000; 1.000.000.000)

alexey-ershkov commented on src/components/Post/Post.js:
```diff
@@ -0,0 +1,128 @@
+ */
+const openReposts = (e) => {
+  const reply = e.currentTarget.parentElement.previousElementSibling;
+  console.log(reply);
```
Консоль логов быть не должно

Similization replied:
Это для себя, потом все уберу

alexey-ershkov commented on src/components/ProfileMenu/ProfileMenu.js:
```diff
@@ -0,0 +1,3 @@
+import { createComponent } from '../../core/models/Component/Component.js';
+
+export const profileMenuComponent = (template) => createComponent(template);
```
Не могу написать к пустому файлу - у вас есть пустые файлы, зачем?

Similization replied:
Это пока заготовка, потом навесим логику

alexey-ershkov commented on src/components/SignupForm/SignupForm.handlebars:
```diff
@@ -0,0 +1,26 @@
+    <button class='btn btn-primary' id='submit' type='submit' class='form__btn__submit'>Sign up</button>
+    <a class='link' href='/login' data-link>Want to sign in instead?</a>
+  </div>
+</form>
```
нет переноса на новую строку в конце

Similization replied:
Добавил

alexey-ershkov reviewed (changes requested):
Посмотрел , отписал, в целом неплохо, но надо доводить до ума, плюс делайте мр с тем, что хотите, чтобы я посмотрел - не обязательно в мастер. Чтобы в диффе было 300-500 строк, но которые я внимательно смогу посмотреть, ибо тут оч много конфигов, верстки и прочего

senago committed:
Add Janitor

senago committed:
Add import

senago committed:
Merge branch 'base-develop' into base

senago committed:
Resolve conflict

Similization committed:
post logic + profile + some fixes

Similization committed:
Merge branch 'base' of https://github.com/frontend-park-mail-ru/2022_1_CJ into base

Similization committed:
profile, components, fixes and active menu buttons

Similization committed:
fixes

senago committed:
Fix linters

senago committed:
Fix linters

senago committed:
Cleanup .dockerignore

senago committed:
Cleanup .dockerignore

senago committed:
Run format

Similization committed:
ccs fixes + signout

Similization committed:
Merge branch 'base' of https://github.com/frontend-park-mail-ru/2022_1_CJ into base

Similization committed:
hot fixes

senago committed:
Start implementing Messenger page

senago committed:
Merge branch 'base-develop' into base

senago committed:
Fix css color var

senago committed:
Fix css color var

senago committed:
Get rid of sources directory

senago committed:
Fix links to static files

senago committed:
Fix conflicts in header

senago committed:
Add error handling and display

Similization committed:
+some fixes
+scripts
+__disabled elements

Similization committed:
correct post
+more __disabled

Similization committed:
+fixes
+more __disabled

Similization committed:
pointer-events: none

senago committed:
Add mocks

senago committed:
Merge branch 'base-stage' into base

senago committed:
Run Linter

Similization committed:
lil profile logic

senago committed:
Bring to live Feed page, fix some bugs

senago committed:
Get rid of boilerplate code

senago committed:
Fix network bug

Similization committed:
settings

Similization committed:
Merge branch 'base' of https://github.com/frontend-park-mail-ru/2022_1_CJ into base

Similization committed:
fixes

senago committed:
Fix ProfileSettings bugs

Similization committed:
+fix scripts
+fix fs-styles
+added commentGroup
+fixed profileMenu
+added default image

senago committed:
Fix data-link behaviour

senago committed:
Fix rendering bug

Similization committed:
+fixes
+post better mock
+new fill data.sh

Similization committed:
Merge branch 'base' of https://github.com/frontend-park-mail-ru/2022_1_CJ into base

Similization committed:
+fill images script corrected
+css fixes
+more webp images
+postGroup recieved time redused
+hide text logic correction
+mock script updates

Similization committed:
hot fixes

Similization committed:
profile more text fixes

senago committed:
Run Format

senago committed:
Cleanup css

Similization committed:
+all datas mock
+css fixes
+handlebars fixes
+added comment Component
+ added lil getProfileLogic (need to be upgarded)
- grid-layout (css)

Similization committed:
+profileSettings fixes

Similization committed:
- __disabled comments

Similization committed:
+50% image horizontalscroll
+warning css
+ errors & alerts fixes

Similization committed:
+fixes
+half image scroll
+half repost logic
+new basic func script
+post likes simplify logic

Similization committed:
+full logic fixes

Similization committed:
+chat component
+fixes
+more icons

Similization committed:
+friends component

Similization committed:
+friends Component
+friendsMenu Component
+ friends View
+ search View
+more icons
+css fixes

senago committed:
Implement some API

senago committed:
Stage

senago committed:
Fix post methods

senago committed:
Add user search in Friends View

senago committed:
Choose better naming

Similization committed:
a lot of fixes
added new icon-a for avatars
and icon stayed for icons
+changed mock datas

Similization committed:
Merge branch 'base' of https://github.com/frontend-park-mail-ru/2022_1_CJ into base

senago committed:
Run lint

Similization committed:
Merge branch 'base' of https://github.com/frontend-park-mail-ru/2022_1_CJ into base

Similization committed:
commented nginx
fixes

senago committed:
Fix nginx config

senago committed:
Fix nginx

senago committed:
Fix import

senago committed:
Add refresh method to Router

senago committed:
Add fallback for webp server

Similization committed:
+lil logic
+lil beaty (describe functions)
profile new post create need to be improved

Similization committed:
Merge branch 'base' of https://github.com/frontend-park-mail-ru/2022_1_CJ into base

Similization committed:
some upload fixes

senago committed:
Fix nginx

Similization committed:
more upload logic

Similization committed:
Merge branch 'base' of https://github.com/frontend-park-mail-ru/2022_1_CJ into base

Similization committed:
+hot fixes
new post upload images need to be improved

Similization committed:
post images upload fixes

senago committed:
Refactor

senago committed:
Run linter

senago committed:
Comment some very nice code

Similization committed:
return friends menu

Similization committed:
delete should delete
